File: algorithms/1436.destination-city/solution.rb
# @param {String[][]} paths
# @return {String}
def dest_city(paths)
  a, b = [], []
  paths.each do |path|
    # departures
    a << path[0]
    # destinations
    b << path[1]
  end
  b.find { |d| !a.include?(d) }
end

File: algorithms/1436.destination-city/solution.rs
use std::collections::HashSet;

impl Solution {
    pub fn dest_city(paths: Vec<Vec<String>>) -> String {
        let mut departures: HashSet<String> = HashSet::new();
        for path in paths.iter() {
            departures.insert(path[0].clone());
        }

        for path in paths.iter() {
            if !departures.contains(&path[1]) {
                return path[1].clone();
            }
        }

        unreachable!()
    }
}
